stdlib/universe/fill.go

```diff
@@ -255,14 +264,33 @@ func (t *fillTransformation) fillTable(w *table.StreamWriter, cr flux.ColReader,
 		return nil
 	}
 	vs := make([]array.Interface, len(w.Cols()))
+
 	for i, col := range w.Cols() {
-		arr := table.Values(cr, i)
-		if i != colIdx {
-			vs[i] = arr
+		var colArr array.Interface
+		// if fillCol is missing AND current col from range is same as fillCol
+		isMissingColCase := colIdx < 0 && col.Label == t.spec.Column
+		if isMissingColCase {
+			colArr = t.addNullColumn(flux.ColumnType(t.spec.Value.Type()), cr.Len())
```

Comment:
This will end up leaking memory because it doesn't release the created column. I think you would be better of refactoring this section because I don't think trying to get it into this loop is working properly. Maybe something like:

```
// Iterate over the existing columns.
for i, col := range cr.Cols() {
    arr := table.Values(cr, i)
    if i != colIdx {
        vs[i] = arr
        vs[i].Retain()
        continue
    }
    vs[i] = t.fillColumn(col.Type, arr, fillValue)
}

// If the fill column is new, create a completely null column and use fill column on it.
if len(w.Cols()) > len(cr.Cols()) {
    typ := flux.ColumnType(t.spec.Value.Type())
    if type == flux.TInvalid {
        // return error indicating the value type is invalid? not sure if this is needed.
    }
    arr := t.addNullColumn(typ, cr.Len())
    defer arr.Release()
    vs[colIdx] = t.fillColumn(typ, arr, fillValue)
}
```

If you do this, you should initialize `colIdx` to be the index of the newly created column.

I think this method simplifies the code. Instead of checking on each iteration if you have the special case, you keep the existing code workflow and fix the new column through an if statement after the initial iteration. The initial iteration has been changed to only iterate over existing columns rather than requiring it to check if the input column is missing. So you only check for the missing column once instead of on each loop.

Reply:
This looks cleaner and easy to read. Thanks for the suggestion, I will update the PR accordingly.